### PR TITLE
Add register logic in register page

### PR DESCRIPTION
**Pull request by AdamJonsson**

### CODE REVIEW

File: src/pages/register-page.tsx [hunk 1]
```diff
@@ -37,17 +50,153 @@ const styles = StyleSheet.create({
...
+            props.router.navigation.replace('Home');
+            return;
+        }
+
+        // No user found, need to create a new one
+        const newUser: User = {
+            authID: userAuthID,
+            address: null,
+            name: null,
+            phone: null,
```
Comment (ellaklara) [inline src/pages/register-page.tsx:166]:
```suggestion
            phone: phoneNumber,
```
Couldn't we add the phone number directly here?

Comment (AdamJonsson) [reply src/pages/register-page.tsx:166]: That true!

**Comment by AdamJonsson** [comment]
Use the AsyncStorage.clear()

File: src/pages/register-page.tsx [hunk 2]
```diff
@@ -37,17 +50,153 @@ const styles = StyleSheet.create({
...
+            );
+
+            // Get a auth user
+            const userAuth = await firebase.auth().signInWithCredential(credential);
+            const userID = userAuth.user?.uid;
+
+            // Saving the user and continue to login
+            if (userID) {
+                await AsyncStorage.setItem(userAuthIDAsyncStorageName, userID);
+                await startLogin(userAuthIDAsyncStorageName);
```
Comment (ellaklara) [inline src/pages/register-page.tsx:140]:
```suggestion
                await startLogin(userID);
```
I think this was the problem!

Comment (AdamJonsson) [reply src/pages/register-page.tsx:140]: Ahh, yea, that looks wrong

**Pull request merged**